src/common/AutoSync.ts

```diff
@@ -12,37 +11,38 @@ import { getServices, Service } from '@models/Service';
 import '@services-apis';
 import { getSyncStore } from '@stores/SyncStore';
 import { PartialDeep } from 'type-fest';
+import browser, { Alarms as WebExtAlarms } from 'webextension-polyfill';
 
 class _AutoSync {
-	isChecking = false;
-	checkTimeoutId: number | null = null;
+	async initFromBackground() {
+		const existingAlarm = await browser.alarms.get('check-auto-sync');
+		if (existingAlarm) {
+			return;
+		}
 
-	init() {
-		void this.check();
-		Shared.events.subscribe('STORAGE_OPTIONS_CHANGE', null, this.onStorageOptionsChange);
+		browser.alarms.create('check-auto-sync', {
+			delayInMinutes: 1,
+			// Check again every hour
+			periodInMinutes: 60,
+		});
 	}
 
-	onStorageOptionsChange = (data: StorageOptionsChangeData) => {
-		if (data.options?.services) {
-			const doCheck = Object.values(data.options.services).some(
-				(serviceValue) =>
-					serviceValue && ('autoSync' in serviceValue || 'autoSyncDays' in serviceValue)
-			);
-			if (doCheck) {
-				void this.check();
-			}
-		}
-	};
+	addBackgroundListeners() {
+		browser.alarms.onAlarm.addListener(this.onAlarm);
+	}
 
-	async check() {
-		if (this.isChecking) {
+	private onAlarm = (alarm: WebExtAlarms.Alarm) => {
+		if (alarm.name !== 'check-auto-sync') {
 			return;
 		}
-		this.isChecking = true;
 
-		if (this.checkTimeoutId !== null) {
-			window.clearTimeout(this.checkTimeoutId);
-		}
+		void this.check();
+	};
+
+	private async check() {
+		await Shared.waitForInit();
+
+		console.log('Checking auto sync...');
```

Comment:
Do you need this in prod, or was it just for debugging?

Reply:
Debugging, will remove.